### PR TITLE
Add complex FE testing in rule & alert filtering

### PR DESCRIPTION
**Pull request by 3nvi**
## Background

This PR adds heavy integration tests for the Rule listing & Alert listing pages, centered around filtering and URLs. While writing those tests I spotted a few bugs and visual issues which were fixed as part of this PR.

Closes https://github.com/panther-labs/panther-enterprise/issues/1191

## Changes

- Wrote tests for ListRule page
- Wrote tests for ListAlerts page
- Fix year format from `YY` to `YYYY`
- Fix `width` of some `Combobox` to account for edge-case values that are big in size
- Replace buggy `_.isEmpty` with something more straightforward

## Testing

- `npm run test`

### CODE REVIEW

File: web/src/pages/ListAlerts/ListAlertBreadcrumbFilters/ListAlertBreadcrumbFilters.tsx [hunk 1]
```diff
@@ -103,7 +103,7 @@ const ListAlertBreadcrumbFilters: React.FC = () => {
                 withPresets
                 withTime
                 variant="solid"
-                format="MM/DD/YY HH:mm"
+                format="MM/DD/YYYY HH:mm"
```
Comment (3nvi) [inline web/src/pages/ListAlerts/ListAlertBreadcrumbFilters/ListAlertBreadcrumbFilters.tsx:106]: changed `10/20/20` to `10/20/2020`

File: web/src/pages/ListAlerts/ListAlertFilters/DropdownFilters.tsx [hunk 2]
```diff
@@ -69,9 +68,8 @@ const DropdownFilters: React.FC = () => {
       } as ListAlertsDropdownFiltersValues),
     [requestParams]
   );
-  const filtersCount = Object.keys(defaultValues).filter(key => !isEmpty(requestParams[key]))
-    .length;
 
+  const filtersCount = Object.keys(defaultValues).filter(key => key in requestParams).length;
```
Comment (3nvi) [inline web/src/pages/ListAlerts/ListAlertFilters/DropdownFilters.tsx:72]: `isEmpty(2)` returns `true` which bugs out the counter. We'll now change this logic and assume that if the filter key is present in the URL params, the filter is considered "active"

File: web/src/pages/ListRules/ListRulesFilters/ListRulesFilters.tsx [hunk 3]
```diff
@@ -166,7 +166,7 @@ const ListRuleFilters: React.FC = () => {
         <Form>
           <FormikAutosave threshold={200} />
           <Flex spacing={4} align="center" pr={4}>
-            <Box width={425}>
+            <Box minWidth={425}>
```
Comment (3nvi) [inline web/src/pages/ListRules/ListRulesFilters/ListRulesFilters.tsx:169]: minor enhancement for large-screen responsiveness

File: web/src/pages/ListRules/ListRulesFilters/ListRulesFilters.tsx [hunk 4]
```diff
@@ -176,7 +176,7 @@ const ListRuleFilters: React.FC = () => {
                 placeholder="Search for a rule..."
               />
             </Box>
-            <Box>
+            <Box minWidth={220}>
```
Comment (3nvi) [inline web/src/pages/ListRules/ListRulesFilters/ListRulesFilters.tsx:179]: this sort combobox has a value of `Most Recently Modified` which requires some space

File: web/src/pages/LogAnalysisOverview/LogAnalysisOverviewBreadcrumbFilters/LogAnalysisOverviewBreadcrumbFilters.tsx [hunk 5]
```diff
@@ -53,7 +53,7 @@ const LogAnalysisOverviewBreadcrumbFilters: React.FC<LogAnalysisOverviewBreadcru
           <Form>
             <FormikAutosave threshold={50} />
             <Flex spacing={4} maxWidth={500}>
-              <Box width={150}>
+              <Box maxWidth={100}>
```
Comment (3nvi) [inline web/src/pages/LogAnalysisOverview/LogAnalysisOverviewBreadcrumbFilters/LogAnalysisOverviewBreadcrumbFilters.tsx:56]: This filter has values `1h`, `3h`, etc., so it doesn't need `150px`. `100px` are more than enough

File: web/src/pages/ListRules/ListRulesFilters/ListRulesFilters.tsx (hunk 3 above)
Comment (alexmylonas) [reply web/src/pages/ListRules/ListRulesFilters/ListRulesFilters.tsx:169]: Should this be also applied to parent component?

File: web/src/pages/ListRules/ListRulesFilters/ListRulesFilters.tsx (hunk 4 above)
Comment (alexmylonas) [reply web/src/pages/ListRules/ListRulesFilters/ListRulesFilters.tsx:179]: Same as above

**Review by alexmylonas — APPROVED** [review]
Solid solid work @3nvi! Small comments but nothing stopping me from approving.

File: web/src/pages/ListRules/ListRulesFilters/ListRulesFilters.tsx (hunk 3 above)
Comment (3nvi) [reply web/src/pages/ListRules/ListRulesFilters/ListRulesFilters.tsx:169]: checked it. Answer is no.
